tests/js/searchtools.spec.js

```diff
@@ -95,7 +95,62 @@ describe('Basic html theme search', function() {
       ]];
       expect(Search.performTermsSearch(searchterms, excluded, terms, titleterms)).toEqual(hits);
     });
+  });
+
+  describe('unicode normalization', function() {
+    it('should find documents indexed with half-width characters using a half-width query', function() {
+      eval(loadFixture("normalization/searchindex.js"));
+
+      const orig = DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION;
+      try {
+        DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION = 'NFKC';
+        [_searchQuery, searchterms, excluded, ..._remainingItems] = Search._parseQuery('Sphinx');
+
+        terms = Search._index.terms;
+        titleterms = Search._index.titleterms;
+
+        hits = [[
+            "index",
+            "Sphinx",
+            "",
+            null,
+            15,
+            "index.rst",
+            "text"],
+        ];
+
+        expect(Search.performTermsSearch(searchterms, excluded, terms, titleterms)).toEqual(hits);
+      } finally {
+        DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION = orig; // restore
+      }
+    });
 
+    it('should find documents indexed with half-width characters using a full-width query', function() {
+      eval(loadFixture("normalization/searchindex.js"));
+
+      const orig = DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION;
+      try {
+        DOCUMENTATION_OPTIONS.SEARCH_UNICODE_NORMALIZATION = 'NFKC';
+        [_searchQuery, searchterms, excluded, ..._remainingItems] = Search._parseQuery('Ｓｐｈｉｎｘ');
```

Comment:
In most cases it makes sense to have exactly one query per unit test; however, because we intend to check equivalence between `Search._parseQuery('sphinx')` and `Search._parseQuery('Ｓｐｈｉｎｘ')`, I would suggest placing that into a separate unit test.

Writing this comment has made me realize: we probably do not need to actually perform a query/retrieve results during that test case.  We can compare the parsed queries for equality.

Reply:
ah, yep . i simplified the test case.

Reply:
Thank you.  Also, my apologies for being unclear in my request; what I would like to suggest is that the final result should contain two test cases:

  1. A test to confirm that document search is able to normalize between query and index (for this, I recommend selecting your test case `should find documents indexed with half-width characters using a full-width query` from 6fb1fb7c48e4747f8062a76c37d727e57bf901e7, because this demonstrates a more-complicated scenario).
  2. A separate test to confirm that a query constructed from full-width text is equivalent to a query constructed from half-width text (for this, I recommend keeping your test case `should parse queries with half-width and full-width characters equivalently` from 20d2749075cca01dafbccbb82f1c5689f681752c).

Reply:
i see. i edited the test case.

Reply:
Thank you.  I think it could be useful to re-add the `should find documents` test case: it demonstrates what the feature does, and also provides a mechanism to catch regressions.